### PR TITLE
ignore host network pods in all endpoints

### PR DESCRIPTION
**Pull request by haouc**
**What type of PR is this?**

bug
**Which issue does this PR fix**:
We shouldn't populate pods in host network to PolicyEndpoint's endpoints.

**What does this PR do / Why do we need it**:
The endpoints from host network pods in ingress/egress/podselector can become redundant data in nodes and CRs. 

**If an issue # is not available please add steps to reproduce and the controller logs**:

**Testing done on this change**:

Tested in local dev cluster and cyclonus tests were passed.

**Automation added to e2e**:

**Will this PR introduce any new dependencies?**:

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by achevuru** [comment]
Let's add a simple test result with host networking pods matching ingress, egress and pod selector endpoint selectors..

File: pkg/resolvers/endpoints.go [hunk 1]
```diff
@@ -491,3 +497,22 @@ func dedupPorts(policyPorts []policyinfo.Port) []policyinfo.Port {
 	}
 	return nil
 }
+
+func (r *defaultEndpointsResolver) getPodList(ctx context.Context, opts client.ListOption) ([]corev1.Pod, error) {
```
Comment (jayanthvn) [inline pkg/resolvers/endpoints.go:501]: Maybe to make this more clear, it can be `getNonHostNetworkingPodList` ?

**Comment by haouc** [comment]
> Let's add a simple test result with host networking pods matching ingress, egress and pod selector endpoint selectors..

The results doc has been shared offline. Please let me know if we want to add the result here as well.

File: pkg/resolvers/endpoints.go (hunk 1 above)
Comment (haouc) [reply pkg/resolvers/endpoints.go]: updated the name.

**Pull request closed without merging**